src/IW/Sync/Search.hs

```diff
@@ -0,0 +1,89 @@
+fetchHaskellIssuesByLabels labels = liftGitHubSearchToApp $ searchHaskellIssuesByLabels labels
+
+-- | Fetch all open issues with Haskell language.
+fetchAllHaskellIssues :: (WithDb env m, WithError m, WithLog env m) => m [GitHub.Issue]
+fetchAllHaskellIssues = fetchHaskellIssuesByLabels []
+
+-- | Lift a github search action to work within our monad stack.
+liftGitHubSearchToApp :: (WithDb env m, WithError m, WithLog env m) => IO (Either GitHub.Error (SearchResult a)) -> m [a]
+liftGitHubSearchToApp githubSearch = do
+    result <- liftIO githubSearch
+    case result of
+        Right (SearchResult count vec) -> do
+            log Info $ "Fetching total of " <> show count <> " results." 
+            return $ Vector.toList vec
+        Left err                       -> throwError . githubErrToAppErr $ err
+
+-- | Search all repositories built with the Haskell language.
+searchHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
+searchHaskellRepos = searchRepos "language:haskell"
+
+-- | Construct a github search query from a list of labels.
+labelsToSearchQuery :: [Label] -> Text
+labelsToSearchQuery = foldMap (\Label{..} -> "label:\"" <> unLabel <> "\" ")
+
+-- | Search for all open Haskell issues with the corresponding labels.
+searchHaskellIssuesByLabels :: [Label] -> IO (Either GitHub.Error (SearchResult GitHub.Issue))
+searchHaskellIssuesByLabels labels = searchIssues $ "language:haskell is:open " <> labelsToSearchQuery labels
+
+-- | Convert a @Maybe Text@ value to a plain @Text@ value.
+maybeTextToText :: Maybe Text -> Text
+maybeTextToText Nothing     = ""
+maybeTextToText (Just text) = text
+
+-- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
+githubRepoToRepo :: GitHub.Repo -> Repo
+githubRepoToRepo githubRepo = Repo{..}
+  where  
+    repoOwner      = RepoOwner . GitHub.untagName . GitHub.simpleOwnerLogin . GitHub.repoOwner $ githubRepo
+    repoName       = RepoName  . GitHub.untagName . GitHub.repoName $ githubRepo
+    repoDescr      = maybeTextToText . GitHub.repoDescription $ githubRepo
+    repoCategories = SqlArray []
+
+-- | Convert a value of the @GitHub.Issue@ type to a value of our own @Issue@ type.
+githubIssueToIssue :: GitHub.Issue -> Issue
+githubIssueToIssue githubIssue = Issue{..}
+  where  
+    issueRepoOwner = undefined
+    issueRepoName  = undefined
+    issueNumber    = GitHub.unIssueNumber . GitHub.issueNumber $ githubIssue
+    issueTitle     = GitHub.issueTitle githubIssue 
+    issueBody      = maybeTextToText . GitHub.issueBody $ githubIssue
+    issueLabels    = SqlArray . Vector.toList $ (\GitHub.IssueLabel{..} -> Label $ GitHub.untagName labelName) <$> GitHub.issueLabels githubIssue 
+
+-- | Extract the @RepoOwner@ from a value of type @GitHub.Issue@.
+parseIssueUrlRepoOwner :: GitHub.Issue -> Maybe RepoOwner
+parseIssueUrlRepoOwner issue = RepoOwner <$> (listToMaybe . take 1 . drop 4 . split ('/'==) . GitHub.getUrl . GitHub.issueUrl) issue 
```

Comment:
I propose to use `stripPrefix` function to get owner and repo. See example from `hit`:

https://github.com/kowainik/hit-on/blob/50c77c051b0e190b1279fac5dcebf34358a09755/src/Hit/Issue.hs#L154-L158

Also, let's have a single function:

```haskell
parseIssueUserData :: Url -> Maybe (RepoOwner, RepoName)
```

Reply:
Awesome :+1: